framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -63,26 +62,34 @@ public ValueTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis)
         atypefactory = analysis.getTypeFactory();
     }
 
+    /**
+     * Returns a list of possible values for {@code subNode}, as casted to a String. Returns null if
+     * {@code subNode}'s type is top/unknown. Returns an empty list if {@code subNode}'s type is
+     * bottom.
+     */
     private List<String> getStringValues(Node subNode, TransferInput<CFValue, CFStore> p) {
         CFValue value = p.getValueOfSubNode(subNode);
-        // @StringVal, @BottomVal, @UnknownVal
-        AnnotationMirror numberAnno =
+        // @StringVal, @UnknownVal, @BottomVal
+        AnnotationMirror stringAnno =
                 AnnotationUtils.getAnnotationByClass(value.getAnnotations(), StringVal.class);
-        if (numberAnno != null) {
-            return AnnotationUtils.getElementValueArray(numberAnno, "value", String.class, true);
+        if (stringAnno != null) {
+            return AnnotationUtils.getElementValueArray(stringAnno, "value", String.class, true);
         }
-        numberAnno = AnnotationUtils.getAnnotationByClass(value.getAnnotations(), UnknownVal.class);
-        if (numberAnno != null) {
-            return new ArrayList<String>();
+        AnnotationMirror topAnno =
+                AnnotationUtils.getAnnotationByClass(value.getAnnotations(), UnknownVal.class);
+        if (topAnno != null) {
+            return null;
         }
-        numberAnno = AnnotationUtils.getAnnotationByClass(value.getAnnotations(), BottomVal.class);
-        if (numberAnno != null) {
-            return Collections.singletonList("null");
+        AnnotationMirror bottomAnno =
+                AnnotationUtils.getAnnotationByClass(value.getAnnotations(), BottomVal.class);
+        if (bottomAnno != null) {
+            return new ArrayList<String>();
```

Comment:
I think the `Collections.singletonList("null")` is returned on purpose to handle `String + null` situation, e.g., `"str" + null == "strnull"`